src/include/migraphx/matcher.hpp

```diff
@@ -844,13 +857,20 @@ auto skip_broadcasts_converts(Ms... ms)
     return skip(name("broadcast", "multibroadcast", "contiguous", "convert"))(ms...);
 }
 
+template <class... Ms>
+auto skip_broadcasts_transposes(Ms... ms)
+{
+    return skip(name("broadcast", "multibroadcast", "contiguous", "transpose"))(ms...);
+}
+
 template <class T>
 inline auto has_value(T x, float tolerance = 1e-6)
 {
     return skip_broadcasts_converts(make_basic_pred_matcher([=](instruction_ref ins) {
-        if(ins->name() != "@literal")
+        if(not ins->can_eval())
```

Comment:
Using can_eval check rather than literal made more sense to me (and is required by the qdq matcher), but if there is a good reason it was coded this way please let me know. 
Doing it this way seems to be breaking just 1 test case in simplify_algebra (simplify_div_const) since that program now gets further simplified than expected by the test case (this extra simplification is valid though).

Reply:
Reverting this change since we dont want to propagate constants before its dedicated pass.